controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -274,3 +289,12 @@ func (a *Adapter) annotateBuildPipelineRunWithSnapshot(pipelineRun *tektonv1beta
 		"snapshot.Name", snapshot.Name)
 	return pipelineRun, nil
 }
+
+func (a *Adapter) removeFinalizerAndContinueProcessing() (controller.OperationResult, error) {
+	if !h.HasPipelineRunFinished(a.pipelineRun) || !controllerutil.ContainsFinalizer(a.pipelineRun, h.BuildPipelineRunFinalizer) {
```

Comment:
do we need this check (`!h.HasPipelineRunFinished(a.pipelineRun)`) within this `if` condition?

Since `removeFinalizerAndContinueProcessing()` is only called within `EnsureSnapshotExists` function that only processes [those PLRs that succeeded](https://github.com/redhat-appstudio/integration-service/pull/357/files#diff-887649ea3498038f46836e7fb616f809a6e09232f7fdec4b833f8d7aa8044db3R67), so this function would always be processing finished (and succeeded) PLRs.

Reply:
We do right now because `removeFinalizerAndContinueProcessing()` is also called if the pipeline has not succeeded yet (I did a find/replace on controllerutils.ContinueProcessing()).  The better solution would be to revert that particular function call and then not rely on this if statement.  It also makes the unit tests simpler.  I'll go change that.